src/ucp/proto/proto_reconfig.c

```diff
@@ -47,10 +59,19 @@ static ucs_status_t ucp_proto_reconfig_progress(uct_pending_req_t *self)
                                   ucp_operation_names, &strb);
         ucs_error("cannot find remote protocol for: %s",
                   ucs_string_buffer_cstr(&strb));
-        ucp_request_complete_send(req, UCS_ERR_CANCELED);
+        ucp_proto_reconfig_abort(req, UCS_ERR_CANCELED);
         return UCS_OK;
     }
 
+    if (ucs_unlikely(ucp_proto_config_is_am(req->send.proto_config) &&
+                     (req->send.msg_proto.am.flags &
+                      UCP_AM_SEND_FLAG_COPY_HEADER))) {
+        status = ucp_proto_am_req_copy_header(req);
+        if (ucs_unlikely(status != UCS_OK)) {
+            return status;
```

Comment:
if copy fails we should abort the operation and return OK from progress. pls see how it's done in other places ucp_proto_am_req_copy_header is used

Reply:
After checking code again we noticed that also for other protocols in protov2 `status != UCS_OK` case is not handled properly. I will fix it and add it to this PR.